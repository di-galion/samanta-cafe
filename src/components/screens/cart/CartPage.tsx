"use client"
import styles from "./styles.module.scss"
import Heading from "@/components/ui/heading/Heading";
import {FC} from "react";
import {BsArrowLeftSquare} from "react-icons/bs";
import {useCart} from "@/hooks/useCart";
import CartItem from "@/components/screens/cart/cart-item/CartItem";
import cn from "classnames";
import Button from "@/components/ui/button/Button";
import {useRouter} from "next/navigation";


const CartPage: FC = () => {
    const {items, total} = useCart()
    const router = useRouter()
    return (
        <div className={styles.cart}>
            <Heading>Корзина</Heading>
            <div className={styles.exit_icon}>
                <BsArrowLeftSquare size={40} />
                <span className={styles.exit_text}>Вернуться в меню</span>
            </div>
            <div className={styles.table}>
                <div className={styles.header}>
                    <div className={styles.col_dish}>
                        Блюдо
                    </div>
                    <div className={cn(styles.col, styles.col_price)}>
                        Цена
                    </div>
                    <div className={styles.col}>
                        Количество
                    </div>
                    <div className={styles.col}>
                        Стоимость
                    </div>
                </div>
                {items && items.map((item) => {
                    return (
                        <CartItem key={item.id} product={item} />
                    )
                })}
            </div>
            <div className={styles.footer}>
                <div className={styles.total_sum}>
                    <div className={styles.text}>Сумма заказа:</div>
                    <div className={styles.text}>{total} руб.</div>
                </div>
                <div className={styles.total}>
                    <div className={styles.text}>Итого:</div>
                    <div className={styles.text} data-testid={"total"}>{total} руб.</div>
                </div>
                <p className={styles.delivery_information}>
                    При заказе от 600 руб. с учетом бонусов и скидок доставка бесплатная.
                    <br/>
                    При заказе менее 600 руб. стоимость доставки 130 руб.
                </p>
            </div>
            <div className={styles.button_container}>
                <Button
                    type={'button'}
                    testId={"button-submit-test"}
                    variant={"red"}
                    onClick={() => router.push('/order')}
                >
                    Оформить заказ
                </Button>
            </div>
        </div>
    )
}

export default CartPage