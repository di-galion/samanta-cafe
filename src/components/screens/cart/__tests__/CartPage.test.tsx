import {renderWithProviders, setupStore} from "@/__tests__/helpers/renderTestApp";
import CartPage from "@/components/screens/cart/CartPage";
import {fireEvent, screen} from "@testing-library/react";
import {act} from "react-dom/test-utils";

const useCartResult = {
    items: [{name: "Product Well", price: 12, category: '3', quantity: 3, slug: "d", images: ["http://local"], id: 1}],
    total: 36
}

const mockedUseRouterPush = jest.fn()
jest.mock('next/navigation', () => ({
    useRouter: () => ({
        push: mockedUseRouterPush
    })
}))

describe("Cart page",  () => {

    afterEach(() => {
        jest.clearAllMocks()
    })

    beforeEach(() => {
        const component = renderWithProviders(<CartPage />, {
            preloadedState: {cart: {items: useCartResult.items,}}})
    })

    test("Cart page render", () => {
        const items = screen.getAllByTestId("cart-item")

        expect(screen.getByTestId("total").textContent).toBe(`${useCartResult.total} руб.`)
        expect(items.length).toBe(useCartResult.items.length)
    })

    test("Should router.push have been called", () => {
        act(() => {
            fireEvent.click(screen.getByTestId("button-submit-test"))
        })

        expect(mockedUseRouterPush).toHaveBeenCalledTimes(1)
    })

})