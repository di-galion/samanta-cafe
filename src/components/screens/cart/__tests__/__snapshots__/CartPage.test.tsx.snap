// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart page Cart page render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cart"
      >
        <h1
          class="heading"
        >
          Оформление заказа
        </h1>
        <div
          class="exit_icon"
        >
          <svg
            fill="currentColor"
            height="40"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 2a1 1 0 0 0-1-1H2a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V2zM0 2a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V2zm11.5 5.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="exit_text"
          >
            Вернуться в меню
          </span>
        </div>
        <div
          class="table"
        >
          <div
            class="header"
          >
            <div
              class="col_dish"
            >
              Блюдо
            </div>
            <div
              class="col col_price"
            >
              Цена
            </div>
            <div
              class="col"
            >
              Количество
            </div>
            <div
              class="col"
            >
              Стоимость
            </div>
          </div>
        </div>
        <div
          class="footer"
        >
          <div
            class="total_sum"
          >
            <div
              class="text"
            >
              Сумма заказа:
            </div>
            <div
              class="text"
            >
              0
               руб.
            </div>
          </div>
          <div
            class="total"
          >
            <div
              class="text"
            >
              Итого:
            </div>
            <div
              class="text"
            >
              0
               руб.
            </div>
          </div>
          <p
            class="delivery_information"
          >
            При заказе от 600 руб. с учетом бонусов и скидок доставка бесплатная.
            <br />
            При заказе менее 600 руб. стоимость доставки 130 руб.
          </p>
        </div>
        <div
          class="button_container"
        >
          <button
            class="red"
            type="button"
          >
            Оформить заказ
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cart"
    >
      <h1
        class="heading"
      >
        Оформление заказа
      </h1>
      <div
        class="exit_icon"
      >
        <svg
          fill="currentColor"
          height="40"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 2a1 1 0 0 0-1-1H2a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V2zM0 2a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V2zm11.5 5.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="exit_text"
        >
          Вернуться в меню
        </span>
      </div>
      <div
        class="table"
      >
        <div
          class="header"
        >
          <div
            class="col_dish"
          >
            Блюдо
          </div>
          <div
            class="col col_price"
          >
            Цена
          </div>
          <div
            class="col"
          >
            Количество
          </div>
          <div
            class="col"
          >
            Стоимость
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="total_sum"
        >
          <div
            class="text"
          >
            Сумма заказа:
          </div>
          <div
            class="text"
          >
            0
             руб.
          </div>
        </div>
        <div
          class="total"
        >
          <div
            class="text"
          >
            Итого:
          </div>
          <div
            class="text"
          >
            0
             руб.
          </div>
        </div>
        <p
          class="delivery_information"
        >
          При заказе от 600 руб. с учетом бонусов и скидок доставка бесплатная.
          <br />
          При заказе менее 600 руб. стоимость доставки 130 руб.
        </p>
      </div>
      <div
        class="button_container"
      >
        <button
          class="red"
          type="button"
        >
          Оформить заказ
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
